app/models/state_file_notification_email.rb

```diff
@@ -16,13 +16,22 @@ class StateFileNotificationEmail < ApplicationRecord
   validates_presence_of :to
   validates_presence_of :body
   validates_presence_of :subject
+  validate :not_unsubscribed
```

Comment:
one of the big challenges of this was thinking of a thorough way to intercept outgoing messages. i landed on having a validation because as long as all other outgoing emails for state file are of this type/model, it should work without having to add more unsubscribe-related code.